close_words.py

```diff
@@ -0,0 +1,181 @@
+import random
+
+def get_text(path):
+    with open(path, 'r', encoding="utf-8") as file:
+        dados = file.read()
+        if dados:
+            return dados
+    return "arquivo vazio"
+
+
+def text_clean(text):
+    """
+    Limpa o texto e garante a similaridade das palavras.
+    :return text: Texto limpo e corrigido.
+    """
+    return (text.replace(",", "")
+            .replace(".", "")
+            .lower()
+            )
+
+
+def get_count(item):
+    """
+    Obtém o segundo elemento (contagem) de cada itém.
+
+    :param item: Uma tupla ou lista de dois elementos onde o segundo elemento é a contagem.
+    :return: A contagem (segundo elemento) do item.
+    """
+    return item[1]
+
+
+def search_word(txt):
+    word = input("digite a palavra que quer pesquisar: ")
+    lista_de_busca = []
+    for i, item in enumerate(text_clean(txt).split()):
+        if item == word:
+            lista_de_busca.append(i)
+    if len(lista_de_busca) > 0:
+        return lista_de_busca
+    return "essa palavra não está no texto"
+
+
+def print_percentage(dict1, dict2):
+    total_itens = sum(dict1.values())
+    total_itens2 = sum(dict2.values())
+    print('-' * 150)
+    print("Porcentagem da frequencia dos anteriores: \n")
+    for item in dict1:
+        percentage_value = (dict1.get(item, 0) / total_itens) * 100
+        print(item, "(", percentage_value, "%) \n")
+    print('-' * 150)
+    print("Porcentagem da frequencia dos posteriores: \n")
+    for item in dict2:
+        percentage_value = (dict2.get(item, 0) / total_itens2) * 100
+        print(item, "(", percentage_value, "%) \n")
+    return 0
+
+
+def desempatar(lista):
+    escolha = random.choice(lista)
+    return escolha
+
+
+def print_most_commons(lista1, lista2, word):
+    if len(lista1) > 1 and len(lista2) > 1:
+        immediate_previous = desempatar(lista1)
+        immediate_subsequent = desempatar(lista2)
+        return f"{immediate_previous} {word} {immediate_subsequent}"
+    elif len(lista1) > 1 >= len(lista2):
+        immediate_previous = desempatar(lista1)
+        immediate_subsequent = lista2[0]
+        return f"{immediate_previous} {word} {immediate_subsequent}"
+    elif len(lista1) <= 1 < len(lista2):
+        immediate_previous = lista1[0]
+        immediate_subsequent = desempatar(lista2)
+        return f"{immediate_previous} {word} {immediate_subsequent}"
+    else:
+        return f"{lista1[0]} {word} {lista2[0]}"
+
+
+def frequency(txt):
```

Comment:
função com mais de 15 linhas pode ser considerada grande.. se uma função for bem definida mesmo e fizer só o que deve, ela não usa muitas linhas.. um função muito longa é mais dificil de manter que varias pequenas (pois vc separada a lógica nomeando cada parte, deixa documentação e a depuração é mais direta)